airflow/api_fastapi/execution_api/routes/task_instances.py

```diff
@@ -243,6 +244,17 @@ def ti_update_state(
             else:
                 updated_state = State.FAILED
         query = query.values(state=updated_state)
+    elif isinstance(ti_patch_payload, TISuccessStatePayload):
+        query = TI.duration_expression_update(ti_patch_payload.end_date, query, session.bind)
+        updated_state = ti_patch_payload.state
+        task_instance = session.get(TI, ti_id_str)
+        TI._register_asset_changes_int(
```

Comment:
Nit/not even quite a nit: If this method is meant to be called from things outside the TI class then it shouldn't be prefixed with `_`

Reply:
Oh and the `int` suffix! That means `internal` but it's not internal anymore.

Reply:
That's right, let me make the changes for that.

I thought `int` meant `in table` lol!

Reply:
This has been fixed